homeassistant/components/sensor/netatmo.py

```diff
@@ -0,0 +1,162 @@
+    """ Get the NetAtmo sensor. """
+
+    try:
+        import lnetatmo
+
+    except ImportError:
+        _LOGGER.exception(
+            "Unable to import lnetatmo. "
+            "Did you maybe not install the package ?")
+
+        return False
+
+    SENSOR_TYPES['temperature'][1] = hass.config.temperature_unit
+    unit = hass.config.temperature_unit
+    authorization = lnetatmo.ClientAuth(config.get(CONF_API_KEY, None),
+                                        config.get('secret_key', None),
+                                        config.get(CONF_USERNAME, None),
+                                        config.get(CONF_PASSWORD, None))
+
+    if not authorization:
+        _LOGGER.error(
+            "Connection error "
+            "Please check your settings for NatAtmo API.")
+        return False
+
+    data = NetAtmoData(authorization)
+
+    dev = []
+    try:
+        # Iterate each module
+        for module_name, monitored_conditions in config['modules'].items():
+            # Test if module exist """
+            if module_name not in data.get_module_names():
+                _LOGGER.error('Module name: "%s" not found', module_name)
+                continue
+            # Only create sensor for monitored """
+            for variable in monitored_conditions:
+                if variable not in SENSOR_TYPES:
+                    _LOGGER.error('Sensor type: "%s" does not exist', variable)
+                else:
+                    dev.append(
+                        NetAtmoSensor(data, module_name, variable, unit))
+    except KeyError:
+        pass
+
+    add_devices(dev)
+
+
+# pylint: disable=too-few-public-methods
+# pylint: disable=too-many-instance-attributes
+class NetAtmoSensor(Entity):
+    """ Implements a NetAtmo sensor. """
+
+    def __init__(self, netatmo_data, module_name, sensor_type, temp_unit):
+        self.client_name = 'NetAtmo'
+        self._name = module_name + '_' + SENSOR_TYPES[sensor_type][0]
+        self.netatmo_data = netatmo_data
+        self.module_name = module_name
+        self.temp_unit = temp_unit
+        self.type = sensor_type
+        self._state = None
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self.update()
+
+    @property
+    def name(self):
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit of measurement of this entity, if any. """
+        return self._unit_of_measurement
+
+    # pylint: disable=too-many-branches
+    def update(self):
+        """ Gets the latest data from NetAtmo API and updates the states. """
+
+        self.netatmo_data.update()
+        data = self.netatmo_data.data[self.module_name]
+
+        if self.type == 'temperature':
+            if self.temp_unit == TEMP_CELCIUS:
```

Comment:
The whole temp unit logic is not necessary. Home Assistant will automatically convert temperature units to the preferred unit.